packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx

```diff
@@ -278,17 +272,30 @@ const CompactInteractiveList = (props) => {
   // map rows differently depending on vertical or horizontal orientation
   const mapRows = () => {
     const placeholdersNumber = isResponsive ? (numberOfRows * numberOfColumns) - rows.length : 0;
-    const result = [...rows];
-    // add placeholder rows to the end.
+    let result = [];
+    if (flowHorizontally) {
+      result = [...rows];
+    } else {
+      for (let i = 0; i < numberOfRows; i += 1) {
+        let x = numberOfColumns - placeholdersNumber;
+        for (let j = i; j < rows.length; j += numberOfRows - (x >= 0 ? 0 : 1)) {
+          if (result.length < rows.length) {
+            result.push(rows[j]);
+            if (x >= 0) { x -= 1; }
+          }
+        }
+      }
+    }
+    // add placeholder rows
     for (let i = rows.length; i < rows.length + placeholdersNumber; i += 1) {
       result.push({ id: `placeholder-row-${i - rows.length + 1}` });
     }
     return result;
   };
 
   const mappedRows = mapRows();
-  const checkIfRowIsLeftMost = (index) => (flowHorizontally ? index % numberOfColumns === 0 : index < rowsPerColumn);
-  const checkIfRowIsTopMost = (index) => (flowHorizontally ? index < numberOfColumns : index % rowsPerColumn === 0);
+  const checkIfRowIsLeftMost = (index) => (index % numberOfColumns === 0);
```

Comment:
That is true that a boolean needs to be named isLeftmost. The problem here is that it is not a boolean, but a method returning boolean. The result of that method call is assigned to a prop called isLeftmost/isRightmost [here](https://github.com/cerner/terra-framework/blob/f475bda3c36cdc50b53afd9dea02ae83fce6f04b/packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx#L337), but I wanted to make sure the method itself is called so that it's clear it is a method, not boolean. Totally agree that the naming isn't good though.